### ISSUE TITLE
2.25.0 is broken on Windows due to ExecutionPolicy requirement; recommended to use 2.24 for now

### ISSUE THREAD

**Issue by jeffmaury** [crc-org/crc#3812]
### General information

  * OS: Windows
  * Hypervisor: Hyper-V
  * Did you run `crc setup` before starting it (Yes/No)?Yes
  * Running CRC on: Laptop

## CRC version
```bash
CRC version: 2.25.0+97705a
OpenShift version: 4.13.6
Podman version: 4.4.4
```
  
## CRC status
```bash
DEBU CRC version: 2.25.0+97705a
DEBU OpenShift version: 4.13.6
DEBU Podman version: 4.4.4
DEBU Running 'crc status'
crc does not seem to be setup correctly, have you run 'crc setup'?
```

## CRC config
```bash
- preset                                : microshift
```

## Host Operating System
```bash

Nom de l'hôte:                              DESKTOP-JEFF
Nom du système d'exploitation:              Microsoft Windows 11 Professionnel
Version du système:                         10.0.22621 N/A build 22621
Fabricant du système d'exploitation:        Microsoft Corporation
Configuration du système d'exploitation:    Station de travail autonome
Type de build du système d'exploitation:    Multiprocessor Free
Propriétaire enregistré:                    Jeff MAURY
Organisation enregistrée:                   
Identificateur de produit:                  00330-80819-37372-AA576
Date d'installation originale:              09/11/2022, 16:39:06
Heure de démarrage du système:              30/08/2023, 10:55:31
Fabricant du système:                       LENOVO
Modèle du système:                          20TJS2F43N
Type du système:                            x64-based PC
Processeur(s):                              1 processeur(s) installé(s).
                                            [01] : Intel64 Family 6 Model 165 Stepping 2 GenuineIntel ~2310 MHz
Version du BIOS:                            LENOVO N2VET29W (1.14 ), 15/03/2021
Répertoire Windows:                         C:\Windows
Répertoire système:                         C:\Windows\system32
Périphérique d'amorçage:                    \Device\HarddiskVolume1
Option régionale du système:                fr;Français (France)
Paramètres régionaux d'entrée:              fr;Français (France)
Fuseau horaire:                             (UTC+01:00) Bruxelles, Copenhague, Madrid, Paris
Mémoire physique totale:                    32 525 Mo
Mémoire physique disponible:                14 954 Mo
Mémoire virtuelle : taille maximale:        65 051 Mo
Mémoire virtuelle : disponible:             47 967 Mo
Mémoire virtuelle : en cours d'utilisation: 17 084 Mo
Emplacements des fichiers d'échange:        C:\pagefile.sys
Domaine:                                    WORKGROUP
Serveur d'ouverture de session:             \\DESKTOP-JEFF
Correctif(s):                               4 Corrections installées.
                                            [01]: KB5028948
                                            [02]: KB5012170
                                            [03]: KB5029263
                                            [04]: KB5028756
Carte(s) réseau:                            5 carte(s) réseau installée(s).
                                            [01]: Intel(R) Wi-Fi 6 AX201 160MHz
                                                  Nom de la connexion : Wi-Fi
                                                  DHCP activé :         Oui
                                                  Serveur DHCP :        192.168.1.254
                                                  Adresse(s) IP
                                                  [01]: 192.168.1.187
                                                  [02]: fe80::5102:6cd:7e09:85b2
                                                  [03]: 2a01:e34:ec7e:3020:5d11:476f:8292:f465
                                                  [04]: 2a01:e34:ec7e:3020:5116:66ca:8d7a:292a
                                            [02]: Bluetooth Device (Personal Area Network)
                                                  Nom de la connexion : Connexion réseau Bluetooth
                                                  État :                Support déconnecté
                                            [03]: Wintun Userspace Tunnel
                                                  Nom de la connexion : OpenVPN Wintun
                                                  État :                Support déconnecté
                                            [04]: TAP-Windows Adapter V9
                                                  Nom de la connexion : OpenVPN TAP-Windows6
                                                  État :                Support déconnecté
                                            [05]: OpenVPN Data Channel Offload
                                                  Nom de la connexion : OpenVPN Data Channel Offload
                                                  État :                Support déconnecté
Configuration requise pour Hyper-V:         Un hyperviseur a été détecté. Les fonctionnalités nécessaires à Hyper-V ne seront pas affichées.

```

### Steps to reproduce

  1. Install 2.25.0
  2. run crc setup
  3. 
  4. 

### Expected
 
Daemon should start

### Actual

Daemon does not start

### Logs

[crc.log](https://github.com/crc-org/crc/files/12474028/crc.log)

Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.

**Comment by anjannath** [comment]
@jeffmaury does running `crc cleanup` then `crc setup` help?

i tried to reproduce by upgrading CRC from `2.24.1` to `2.25.0` and did not face any issues with `crc setup`, daemon was successfully started (installed MSI then ran crc setup)

```
PS C:\Users\anath> crc.exe setup --log-level debug
DEBU CRC version: 2.25.0+97705a
DEBU OpenShift version: 4.13.6
DEBU Podman version: 4.4.4
DEBU Running 'crc setup'
INFO Using bundle path C:\Users\anath\.crc\cache\crc_podman_hyperv_4.4.4_amd64.crcbundle
INFO Checking minimum RAM requirements
DEBU Total memory of system is 34248990720 bytes
INFO Checking if current user is in crc-users and Hyper-V admins group
DEBU Running '(Get-LocalGroupMember -Group 'crc-users').Name'
DEBU Checking current user is in the 'crc-user' group
DEBU group members: DESKTOP-J8H86J0\anath
DEBU Running '(Get-LocalGroupMember -SID 'S-1-5-32-578').Name'
DEBU Checking current user is in the 'Hyper-v Administrators' group
DEBU group members: DESKTOP-J8H86J0\anath
INFO Checking if CRC bundle is extracted in '$HOME/.crc'
INFO Checking if C:\Users\anath\.crc\cache\crc_podman_hyperv_4.4.4_amd64.crcbundle exists
DEBU C:\Users\anath\.crc\cache\crc_podman_hyperv_4.4.4_amd64.crcbundle exists
INFO Checking if the daemon task powershell script is present
INFO Checking if the daemon task is installed
DEBU Running 'Get-ScheduledTask -TaskName crcDaemon'
DEBU Running '(Get-ScheduledTask -TaskName "crcDaemon").Version'
DEBU expected crcDaemon task to be on version '2.25.0' but got '2.24.1
'
INFO Installing the daemon task
DEBU Running 'Get-ScheduledTask -TaskName crcDaemon'
DEBU Running '(Get-ScheduledTask -TaskName "crcDaemon").State'
DEBU Running 'Get-ScheduledTask -TaskName crcDaemon'
DEBU Running '(Get-ScheduledTask -TaskName "crcDaemon").Version'
DEBU Running 'Unregister-ScheduledTask -TaskName crcDaemon -Confirm:$false'
DEBU Running 'Register-ScheduledTask -Xml '<?xml version="1.0" encoding="UTF-16"?>
<Task version="1.3" xmlns="http://schemas.microsoft.com/windows/2004/02/mit/task">
  <RegistrationInfo>
    <Description>Run crc daemon as a task</Description>
    <Version>2.25.0</Version>
  </RegistrationInfo>
  <Settings>
    <DisallowStartIfOnBatteries>false</DisallowStartIfOnBatteries>
    <StopIfGoingOnBatteries>false</StopIfGoingOnBatteries>
    <Hidden>true</Hidden>
    <MultipleInstancesPolicy>IgnoreNew</MultipleInstancesPolicy>
    <IdleSettings>
      <Duration>PT10M</Duration>
      <WaitTimeout>PT1H</WaitTimeout>
      <StopOnIdleEnd>true</StopOnIdleEnd>
      <RestartOnIdle>false</RestartOnIdle>
    </IdleSettings>
    <UseUnifiedSchedulingEngine>true</UseUnifiedSchedulingEngine>
    <ExecutionTimeLimit>PT0S</ExecutionTimeLimit>
  </Settings>
  <Triggers>
    <LogonTrigger>
      <UserId>DESKTOP-J8H86J0\anath</UserId>
    </LogonTrigger>
  </Triggers>
  <Actions Context="Author">
    <Exec>
      <Command>powershell.exe</Command>
      <Arguments>-WindowStyle Hidden -Command &amp; &#39;C:\Users\anath\.crc\bin\hidden_daemon.ps1&#39;</Arguments>
    </Exec>
  </Actions>
</Task>
' -TaskName crcDaemon'
INFO Checking if the daemon task is running
DEBU Running '(Get-ScheduledTask -TaskName "crcDaemon").State'
DEBU expected crcDaemon task to be in 'Running' but got 'Ready
'
INFO Running the daemon task
DEBU Running 'Start-ScheduledTask -TaskName crcDaemon'
DEBU retry loop: attempt 0
DEBU error: Temporary error: daemon is not running yet - sleeping 2s
DEBU retry loop: attempt 1
INFO Checking admin helper service is running
DEBU Running '(Get-Service crcAdminHelper).Status'
Your system is correctly setup for using CRC. Use 'crc start' to start the instance

PS C:\Users\anath> Get-ScheduledTask -TaskName 'crcDaemon'

TaskPath                                       TaskName                          State
--------                                       --------                          -----
\                                              crcDaemon                         Running

```

**Comment by jeffmaury** [comment]
Check the timestamp the file is huge i'm a long time user of CRC ! 🥲

**Comment by jeffmaury** [comment]
Please note that it has also been reported by a Podman Desktop user on Discord: https://discord.com/channels/852634929845239818/1085121643782873118/threads/1145952685464485938

**Comment by anjannath** [comment]
A `crc cleanup` did not help? I only face this issue, when there's another instance of the daemon running. Make sure there are no stray crc processes, if there are please kill those from the task manager and then try to run `crc setup` again.

I went through the discord thread, it seems that user tried to start the daemon manually and then run `crc setup` which will not work, there can be only one instance of the daemon running, so the scheduled task will just start and end immidiately, and the state will be "Ready" instead of "Running"

```
PS C:\Users\anath> crc.exe daemon --log-level debug
DEBU CRC version: 2.25.0+97705a
DEBU OpenShift version: 4.13.6
DEBU Podman version: 4.4.4
DEBU Running 'crc daemon'
daemon is already running
```

@jeffmaury can you please also share the content of `~\.crc\bin\hidden_daemon.ps1`

**Comment by jeffmaury** [comment]
crc cleanup did not help:

![image](https://github.com/crc-org/crc/assets/695993/cf8e2e1d-750d-41ec-b56e-288c396adb70)

**Comment by anjannath** [comment]
So after debugging with @jeffmaury we found the cause to be the `ExecutionPolicy` which was set to `Restricted`. 

To fix the issue we ran `Set-ExecutionPolicy -ExecutionPolicy RemoteSigned` and then `crc setup` successfully started the daemon process

@gbraad maybe we need to add a preflight check or custom action in the MSI to set the execution policy to allow to run posh scripts?

**Comment by slemeur** [comment]
Why was it setup to "restricted"? shouldn't be an error message explaining the error too?

**Comment by jeffmaury** [comment]
This is the default on Windows

**Comment by gbraad** [comment]
yep; this is the default... and most likely why I can not repeat the issue: I have changed this policy :-/

**Comment by gbraad** [comment]
perhaps a GUI wrapper to handle this case might be cleaner (changing the policy might not be something that everybody can do).

**Comment by djg2018** [comment]
Hi, Im trying to install CRC on my laptop, but the issue mentioned above never got resolve. I've followed this whole issue thread and did every step mentioned still the issue remains same as shown below. 

```
PS C:\Users\dgun> crc setup --log-level debug
DEBU CRC version: 2.37.1+36d451
DEBU OpenShift version: 4.15.14
DEBU Running 'crc setup'
INFO Using bundle path C:\Users\dgun\.crc\cache\crc_hyperv_4.15.14_amd64.crcbundle
INFO Checking minimum RAM requirements
DEBU Total memory of system is 68458733568 bytes
INFO Check if Podman binary exists in: C:\Users\dgun\.crc\bin\oc
INFO Checking if running in a shell with administrator rights
DEBU Running '$currentPrincipal = New-Object Security.Principal.WindowsPrincipal([Security.Principal.WindowsIdentity]::GetCurrent());$currentPrincipal.IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)'
INFO Checking Windows release
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion" -Name ReleaseId).ReleaseId'
INFO Checking Windows edition
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").EditionID'
DEBU Running on Windows Professional edition
INFO Checking if Hyper-V is installed and operational
DEBU Running '@(Get-Wmiobject Win32_ComputerSystem).HypervisorPresent'
DEBU Running '@(Get-Service vmms).Status'
INFO Checking if Hyper-V service is enabled
DEBU Running '@(Get-Service vmms).Status'
INFO Checking if crc-users group exists
DEBU Running 'Get-LocalGroup -Name crc-users'
INFO Checking if current user is in crc-users and Hyper-V admins group
DEBU Running '(Get-LocalGroupMember -Group 'crc-users').Name'
DEBU Checking current user is in the 'crc-user' group
DEBU group members: XXX\dgun
DEBU Running '(Get-LocalGroupMember -SID 'S-1-5-32-578').Name'
DEBU Checking current user is in the 'Hyper-v Administrators' group
DEBU group members: XXXXXXXX\Administrator,DHEERAJ\DefaultAccount,DHEERAJ\dgun
INFO Checking if vsock is correctly configured
DEBU Running 'Get-Item -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Virtualization\GuestCommunicationServices\00000400-FACB-11E6-BD58-64006A7986D3"'
INFO Checking if CRC bundle is extracted in '$HOME/.crc'
INFO Checking if C:\Users\dgun\.crc\cache\crc_hyperv_4.15.14_amd64.crcbundle exists
DEBU C:\Users\dgun\.crc\cache\crc_hyperv_4.15.14_amd64.crcbundle exists
INFO Checking if the win32 background launcher is installed
DEBU Running '(Get-Item 'C:\Program Files\Red Hat OpenShift Local\crc-background-launcher.exe').VersionInfo.FileVersion'
DEBU Found crc-background-launcher.exe version 0.0.0.1
INFO Checking if the daemon task is installed
DEBU Running 'Get-ScheduledTask -TaskName crcDaemon'
DEBU Running '(Get-ScheduledTask -TaskName "crcDaemon").Version'
INFO Checking if the daemon task is running
DEBU Running '(Get-ScheduledTask -TaskName "crcDaemon").State'
DEBU expected crcDaemon task to be in 'Running' but got 'Ready
'
INFO Running the daemon task
DEBU Running 'Start-ScheduledTask -TaskName crcDaemon'
DEBU retry loop: attempt 0
DEBU error: Temporary error: daemon is not running yet - sleeping 2s
DEBU retry loop: attempt 1
DEBU error: Temporary error: daemon is not running yet - sleeping 2s
DEBU retry loop: attempt 2
DEBU error: Temporary error: daemon is not running yet - sleeping 2s
DEBU retry loop: attempt 3
DEBU error: Temporary error: daemon is not running yet - sleeping 2s
DEBU retry loop: attempt 4
DEBU error: Temporary error: daemon is not running yet - sleeping 2s
DEBU retry loop: attempt 5
DEBU error: Temporary error: daemon is not running yet - sleeping 2s
DEBU retry loop: attempt 6
DEBU error: Temporary error: daemon is not running yet - sleeping 2s
DEBU retry loop: attempt 7
DEBU error: Temporary error: daemon is not running yet - sleeping 2s
DEBU RetryAfter timeout after 8 tries
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion" -Name ReleaseId).ReleaseId'
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").EditionID'
Temporary error: daemon is not running yet (x8)
```

**Comment by anjannath** [comment]
@djg2018 could you please open a new issue by providing the information asked in the issue template
